File: Livro - Módulo Java/4 - Estrutura de Repetição/Exercicio24.java
import java.util.Scanner;

public class Exercicio24 {
    public static void main(String[] args) {
        // Será digitado uma série de números inteiros. Faça um algoritmo para calcular
        // e imprimir o produto desses valores, isto é, o resultado da multiplicação
        // de todos os números. Número de entrada desconhecido.

        Scanner scanner = new Scanner(System.in);
        int produto = 1;

        while (true) {
            System.out.print("Digite um número inteiro (ou pressione 0 para encerrar): ");
            int entrada = scanner.nextInt();

            if (entrada == 0) {
                break;
            }

            produto *= entrada;
            System.out.println("Resultado: " + produto);
        }
    }
}

File: Livro - Módulo Python/4 - Estrutura de Repetição/Exercicio.24.py
#24. Será digitado uma série de números inteiros. Faça um algoritmo para calcular e imprimir o produto desses
# valores, isto é, o resultado da multiplicação de todos os números. Número de entrada desconhecido.

soma = 1
while True:
    entrada = int(input("Digite um número inteiro (ou pressione 0 para encerrar): "))
    if entrada == 0:
        break
    soma *= entrada
print(f"Resultado: {soma}")